airgun/entities/contenthost.py

```diff
@@ -39,7 +41,6 @@ def read_all(self, widget_names=None):
     def read(self, entity_name, widget_names=None):
         """Read content host details, optionally read only the widgets in widget_names."""
         view = self.navigate_to(self, 'Edit', entity_name=entity_name)
-        view.is_displayed
```

Comment:
have you tried using below code earlier. we might be not needed to so that whole stuff.   

```python
       view.wait_displayed()
       self.browser.plugin.ensure_page_safe()
```

Reply:
In this [comment](d9f2c1bedad625bfe8a7839b205a21de9918f150) you said that you do not want to have it in the read method and it should be moved to the navigation steps and now it is fine to have it in the read method? I am confused.

Reply:
```python 
       view.wait_displayed()
       self.browser.plugin.ensure_page_safe()
```
you could have multiple approaches to solve the problem, If the above code does not work for you then I'm fine with the current approach.

Reply:
I should have mentioned earlier, come to know the later part. That's fine, thanks for doing those changes.